bemserver_ui/static/scripts/modules/views/analysis/weather.js

```diff
@@ -0,0 +1,330 @@
+                    option.selected = true;
+                }
+                this.#periodMonthSelectElmt.appendChild(option);
+            }
+
+            this.#periodDaySelectElmt.innerHTML = "";
+            const daysInMonth = TimeCalendar.getDaysInMonth(this.#yearRef, this.#monthRef);
+            for (let day = 1; day <= new Date(this.#yearRef, daysInMonth, 0).getDate() ; day++) {
+                let option = document.createElement("option");
+                option.value = day;
+                option.textContent = day;
+                if(day == this.#dayRef){
+                    option.selected = true;
+                }
+                this.#periodDaySelectElmt.appendChild(option);
+            }
+        }
+
+        if (this.#periodTypeSelectElmt.value.startsWith("Month")) {
+            if (this.#previousPeriodType == null) {
+                this.#periodMonthSelectElmt.value = this.#monthRef.toString();
+            }
+            this.#periodMonthSelectElmt.classList.remove("d-none", "invisible");
+        }
+        else {
+            this.#periodMonthSelectElmt.classList.add("d-none", "invisible");
+        }
+
+        if (this.#periodTypeSelectElmt.value.startsWith("Day") || this.#periodTypeSelectElmt.value.startsWith("Week")) {
+            if (this.#previousPeriodType == null) {
+                this.#periodDaySelectElmt.value = this.#dayRef.toString();
+            }
+            this.#periodDaySelectElmt.classList.remove("d-none", "invisible");
+            this.#periodMonthSelectElmt.classList.remove("d-none", "invisible");
+        }
+        else {
+            this.#periodDaySelectElmt.classList.add("d-none", "invisible");
+        }
+
+        if (this.#periodTypeSelectElmt.value.startsWith("Last") ) {
+            this.#periodYearSelectElmt.classList.add("d-none", "invisible");
+            this.#forecastWrapper.classList.remove("d-none", "invisible");
+        }
+        else {
+            this.#periodYearSelectElmt.classList.remove("d-none", "invisible");
+            this.#forecastWrapper.classList.add("d-none", "invisible");
+            this.#forecast.checked = false;
+        }
+
+        this.#previousPeriodType = this.#periodTypeSelectElmt.value;
+        this.#previousDaySelected = this.#periodDaySelectElmt.value;
+        this.#previousYearSelected = this.#periodYearSelectElmt.value;
+    }
+
+    #updateDaysInMonth() {
+        const daysInMonth = TimeCalendar.getDaysInMonth(this.#periodYearSelectElmt.value, this.#periodMonthSelectElmt.value);
+        this.#periodDaySelectElmt.innerHTML = "";
+
+        for (let day = 1; day <= daysInMonth; day++) {
+            let option = document.createElement("option");
+            option.value = day;
+            option.textContent = day;
+            if(day == this.#previousDaySelected){
+                option.selected = true;
+            }
+            this.#periodDaySelectElmt.appendChild(option);
+        }
+    }
+
```

Comment:
Add a function, maybe `#updateTsInfoModal(name, dataset, forecast, is_admin)`, to update modal content (move here the code written in `TimeseriesChartWeather.#showTSInfo`, more or less).